dusty/systems/virtualbox/__init__.py

```diff
@@ -131,3 +141,29 @@ def get_host_ip():
             if line.startswith('IPAddress'):
                 return line.split()[1]
     raise RuntimeError('Host IP in host only network not found')
+
+def _file_exists_on_vm(file_path):
+    result = _run_command_on_vm('if [ -f {} ]; then echo yes; else echo no; fi'.format(file_path),
+                                return_output=True).rstrip()
+    if result == 'yes':
+        return True
+    return False
+
+def required_absent_assets(assembled_specs):
+    required_and_absent = []
+    for asset_key, asset_info in assembled_specs['assets'].iteritems():
+        if asset_info['required_by'] and not asset_is_set(asset_key):
+            required_and_absent.append(asset_key)
+    return required_and_absent
+
+def asset_vm_path(asset_key):
+    return os.path.join(constants.VM_ASSETS_DIR, asset_key)
+
+def asset_is_set(asset_key):
+    return _file_exists_on_vm(asset_vm_path(asset_key))
```

Comment:
I think we probably want to batch this. Doing a bunch of individual SSH commands will get slow.